### PR TITLE
Test against allowed failure Capybara master

### PR DESCRIPTION
**Pull request by twalpole**
This adds testing against Capybara master to see potentially upcoming failures, it also installs the latest bundler when running tests to fix the "NoMethodError: undefined method `spec' for nil:NilClass" and updates the travis config to use the container mode for faster test initiation.

### CODE REVIEW

**Comment by mhoran** [comment]
The apt add-on is great! We had a slight optimization in there to speed up installation for the Qt 4 builds, but I think this is less of an issue these days. Also, capybara-webkit 2.0 will only support Qt 5, so the conditional logic is going away.

This looks good. The only thing is it'd be nice to have our build trigger when capybara master is bumped. Good to see it's great for now, though!

**Comment by twalpole** [comment]
@mhoran Yeah -- being able to specify a dependency whose update triggers a new build would be great, unfortunately I don't think Travis supports that currently (at least I've never seen anything about it - happy to be proven wrong)

File: lib/capybara/webkit/driver.rb [hunk 1]
```diff
@@ -287,7 +287,8 @@ def clear_cookies
     end
 
     def invalid_element_errors
-      [Capybara::Webkit::ClickFailed]
+      [Capybara::Webkit::ClickFailed,
+       Capybara::Webkit::NodeNotAttachedError]
```
Comment (jferris) [inline lib/capybara/webkit/driver.rb]: Is this change related to the switch to testing against the master branch?

Comment (twalpole) [inline lib/capybara/webkit/driver.rb]: @jferris Yes ---  It happens to be treated by the main synchronize loop in Capybara as an invalid element error because it's derived from Capybara::ElementNotFound, but it really should be returned in #invalid_element_errors which a fixing of some tests in Capybara showed.

**Comment by jferris** [comment]
Thanks! I merged this into master.
